drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/CommonOnMoveHandler.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.handlers;
+
+import java.util.Optional;
+
+import javax.enterprise.context.Dependent;
+
+import com.ait.lienzo.client.core.types.Point2D;
+import org.drools.scenariosimulation.api.model.FactMapping;
+import org.drools.scenariosimulation.api.model.FactMappingValue;
+import org.drools.scenariosimulation.api.model.FactMappingValueStatus;
+import org.drools.scenariosimulation.api.model.Scenario;
+import org.drools.workbench.screens.scenariosimulation.client.events.SetGridCellValueEvent;
+import org.drools.workbench.screens.scenariosimulation.client.metadata.ScenarioHeaderMetaData;
+import org.drools.workbench.screens.scenariosimulation.client.popover.ErrorReportPopoverPresenter;
+import org.drools.workbench.screens.scenariosimulation.client.popover.PopoverView;
+import org.drools.workbench.screens.scenariosimulation.client.resources.i18n.ScenarioSimulationEditorConstants;
+import org.drools.workbench.screens.scenariosimulation.client.utils.ScenarioSimulationUtils;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridColumn;
+import org.uberfire.ext.wires.core.grids.client.model.GridColumn;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+/**
+ * This class is meant to provide common implementations for <b>on hover</b> behavior to be used by both mouse keyboard handler
+ */
+@Dependent
+public class CommonOnMoveHandler extends AbstractScenarioSimulationGridPanelHandler {
+
+    /* This parameter must be synchronized with POPOVER_WIDTH static variable in ErrorReportPopoverView.less */
+    private static int POPOVER_WIDTH = 200;
+    private static String NULL = "null";
+
+    protected ErrorReportPopoverPresenter errorReportPopupPresenter;
+
+    protected Integer currentlyShownHeaderRowIndex = -1;
+    protected Integer currentlyShownHeaderColumnIndex = -1;
```

Comment:
`currentlyShownHeaderRowIndex` and `currentlyShownHeaderColumnIndex` are never used, is it intentional?

Reply:
At the moment these are not used, you right. Could be used in the future when a popover for the header cells is required. But I agree with you, they can be removed. 
Done!